src/base.js

```diff
@@ -364,14 +396,59 @@ export default class KintoClientBase {
       return raw ? { json: msg, headers: { get() {} } } : msg;
     }
     await this.fetchServerSettings();
-    const result = await this.http.request(this.remote + request.path, {
-      ...request,
-      body: stringify ? JSON.stringify(request.body) : request.body,
-    });
+    // FIXME: Because any fetch() option is accepted here, `request`
+    // could have lots of parameters that wouldn't be accepted as part
+    // of a batch request. This makes it easier to shoot yourself in
+    // the foot. We should probably only pass through the four
+    // parameters accepted in a batch request: method, body, path, and
+    // headers.
```

Comment:
Several options here:

```js
// 1. the imperative way

const {method, headers, body} = request;
const finalRequest = {};
if (method) {
  finalRequest.method = method;
}

// 2. the costly serialization way

const {method, headers, body} = request;
// stringify will drop undefined properties
const finalRequest = JSON.parse(JSON.stringify({method, headers, body}));

// 3. the dedicated utility way
const {method, headers, body} = request;
function cleanUndefinedProperties(obj) {
  const result = {};
  for (const key in obj) {
    if (typeof obj[key] !== undefined) {
      result[key] = obj[key];
    }
  }
  return result;
}
const finalRequest = cleanUndefinedProperties(request);
```

None are great, so I don't know. I think I could live with undefined request properties passed to `request` anyway, as I think `fetch` implementations all deal just fine with this situation.